lib/Data/Chronicle/Reader.pm

```diff
@@ -105,9 +112,16 @@ sub get {
     my $name     = shift;
 
     my $key         = $category . '::' . $name;
-    my $cached_data = $self->cache_reader->get($key);
 
-    return JSON::from_json($cached_data) if defined $cached_data;
+    if ( defined $self->cache_reader ) {
+        my $cached_data = $self->cache_reader->get($key);
+        return JSON::from_json($cached_data) if defined $cached_data;
+    }
+
+    if ( defined $self->memory_map ) {
+        return $self->memory_map->{$key};
+    }
```

Comment:
This might work to get rid of chronicle_reader usage. But I think this is just another layer of complexity added to the code. The whole structure of quant framework doesn't change. We are covering up the flaws. Now the `memory_map` is no different than `chronicle_reader` in my opinion. Maybe this is the right approach but I think we should really stop and think about this.

I believe we our ultimate goal is: 

```
PricingEngine->new(
   volsurface_data => $vsd,
   quoted_currency_term_structure => $qcts,
   asset_term_structure => $ats,
   holidays => $holiday_data,
   etc ...
);
```

Since the whole quant framework architecture is still a little convoluted, we can't achieve this by not passing in `chronicle_reader` because `VolSurface::Delta` object for example, will need access to `TradingCalendar` when it calculates volatility.

We could have QuantFramework rewritten in another way where:

```
my $qf = QuantFramework->new(
   volsurface_data => $vsd,
   holiday_data => $holiday_data
);

$qf->get_volatility({...})

#And in subroutine that gets volatility

sub get_volatility {
   my $self = shift;

   my $trading_calendar = Quant::Framework::TradingCalendar->(holiday_data => $holiday_data);
   return Quant::Framework::VolSurface->new(
      volsurface_data => $self->volsurface_data,
      trading_calendar => $trading_calendar,
   )->get_volatility({...});
}
```

This way it ensures single point of access to our Quant::Framework model and it gets rid of all the recursion that we have in the code. I have pointed out this before but it was brushed off with the reason being the refactoring will come later.